packages/state-transition/src/util/seed.ts

```diff
@@ -73,7 +83,9 @@ export function computeProposerIndex(
       return candidateIndex;
     }
     i += 1;
-    if (i === indices.length) {
+    // Spec does not have this condition to end infinite loop. Spec test won't pass if we
+    // only loop the validator indices once. Electra spec test requires MAX_ITERATION_OVER_VALIDATORS >= 6
+    if (i === indices.length * MAX_ITERATION_OVER_VALIDATORS) {
```

Comment:
In the original spec, this loop is infinite where as Lodestar has a early return statement to return -1 after looping through the validator set once. 

Lodestar will return -1 for proposer index when the validator set is small. This phenomenon is surfaced during spec testing. Tests will fail unless we loop the validator set at least 6 times.

As a workaround, I have introduced `MAX_ITERATION_OVER_VALIDATORS` to relax the early return condition, but would also consider to remove it altogether to align with the spec [here](https://github.com/ethereum/consensus-specs/blob/389b2ddfb954731da7ccf4c0ef89fab2d4575b99/specs/electra/beacon-chain.md?plain=1#L445)

Reply:
I think it's not correct to do early return after an iteration, the same validator index could have different `randByte` value after each iteration
this could be read as an improvement we made compared to spec, but it's important to be spec compliant first
suggest removing this condition completely